exercises/practice/bowling/src/lib.rs

```diff
@@ -4,18 +4,76 @@ pub enum Error {
     GameComplete,
 }
 
-pub struct BowlingGame {}
+const FULL_PINS: u16 = 10;
+const NUMBER_OF_FRAMES: usize = 10;
+const TWO_THROWS: usize = 2;
+
+pub struct BowlingGame {
+    throws: Vec<u16>,
+    first_throw: bool,
+}
 
 impl BowlingGame {
     pub fn new() -> Self {
-        unimplemented!();
+        BowlingGame {
+            throws: Vec::new(),
+            first_throw: true,
+        }
     }
 
     pub fn roll(&mut self, pins: u16) -> Result<(), Error> {
```

Comment:
Although the logic is correct, the code is messy and not easy to follow at first.  Break long if-s, using arguments. Space between different logic parts and add single-line document before each one.